core/arch/riscv/mm/core_mmu_arch.c

```diff
@@ -300,12 +301,26 @@ static void core_init_mmu_prtn_tee(struct mmu_partition *prtn,
 	size_t n = 0;
 	void *pgt = core_mmu_get_root_pgt_va(prtn);
 
+	/* Clear table before using it. */
+	memset(prtn->root_pgt, 0, RISCV_MMU_PGT_SIZE * CFG_TEE_CORE_NB_CORE);
 	memset(pgt, 0, RISCV_MMU_PGT_SIZE);
 	memset(prtn->pool_pgts, 0, RISCV_MMU_MAX_PGTS * RISCV_MMU_PGT_SIZE);
 
 	for (n = 0; !core_mmap_is_end_of_table(mm + n); n++)
 		if (!core_mmu_is_dynamic_vaspace(mm + n))
 			core_mmu_map_region(prtn, mm + n);
+
+	/*
+	 * Primary mapping table is ready at index `get_core_pos()`
+	 * whose value may not be ZERO. Take this index as copy source.
+	 */
+	for (n = 0; n < CFG_TEE_CORE_NB_CORE; n++) {
+		if (n == get_core_pos())
+			continue;
+
+		memcpy(&prtn->root_pgt[n], &prtn->root_pgt[get_core_pos()],
+		       RISCV_MMU_PGT_SIZE);
+	}
```

Comment:
suggestion:
```c
	for (n = 0; n < CFG_TEE_CORE_NB_CORE; n++)
		if (n != get_core_pos())
			prtn->root_pgt[n] = prtn->root_pgt[get_core_pos()];

Reply:
We must use `memcpy()` instead of direct pointer assignment, since we are copying page table entries.

Reply:
`RISCV_MMU_PGT_SIZE` is `sizeof(struct mmu_pgt)` so the `memcpy()` instruction copies a single cell of `prtn->root_pg`, no?

Reply:
> `RISCV_MMU_PGT_SIZE` is `sizeof(struct mmu_pgt)` so the `memcpy()` instruction copies a single cell of `prtn->root_pg`, no?

Yes, it copies  a single cell like what ARM does also (https://github.com/OP-TEE/optee_os/blob/master/core/arch/arm/mm/core_mmu_lpae.c#L641)

Reply:
Changed in https://github.com/OP-TEE/optee_os/pull/6713/commits/79246e385fd59a63047ac7d201c1d8ab75196c31

Reply:
I still think this implementation is equivalent to 
`prtn->root_pgt[n] = prtn->root_pgt[get_core_pos()]`
which is easier to understand. See `#define RISCV_MMU_PGT_SIZE	(sizeof(struct mmu_pgt))`.

The Arm implementation copies several MMU translation cells.